# First DS Assignment with Python

## Introduction
America League Baseball is the highest level of professional baseball in the United States and Canada, consisting of teams that compete in the American League (AL) and the National League (NL). Each league is further divided into three divisions: East, Central, and West.

## Libraries and Packages
These are the list of Python libraies to use:
- Pandas
- Numpy
- Matplotlib
- Seaborn

## Source of dataset
The DS was gotten from
[America League Base Ball Data](https://en.wikipedia.org/wiki/American_League)

### Import necessary libraries

In [190]:
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns

In [191]:
pd1 = pd.read_csv('albb-salaries-2003.csv')

In [192]:
# take a look at the dataset
pd1.head()

Team           Player      Salary Position
0  New York Yankees    Acevedo, Juan     900,000  Pitcher
1  New York Yankees  Anderson, Jason     300,000  Pitcher
2  New York Yankees   Clemens, Roger  10,100,000  Pitcher
3  New York Yankees  Contreras, Jose   5,500,000  Pitcher
4  New York Yankees   Flaherty, John     750,000  Catcher

In [193]:
pd1.shape # checking the size of the data

(381, 4)

In [194]:
pd1.describe()

Team         Player   Salary Position
count                 381            381      381      381
unique                 14            381      172        7
top     Cleveland Indians  Acevedo, Juan  300,000  Pitcher
freq                   31              1       44      180

### Question 1
**How many pitchers does the New York Yankees have?**

In [195]:
len(pd1[(pd1["Position"] == "Pitcher") & (pd1["Team"] == "New York Yankees")])

14

After filtering down the position to be pitcher and the team to be new york yankees, the lenght of the new york yankees pitcher is **14**

### Question 2
**Who is (are) the highest-paid player(s) in the Cleveland Indians Team?**

In [202]:
clevelandI_team = pd1[pd1['Team'] == "Cleveland Indians"]

In [203]:
clevelandI_team

Team               Player     Salary        Position  \
82   Cleveland Indians      Anderson, Brian  1,500,000         Pitcher   
83   Cleveland Indians          Baez, Danys  5,125,000         Pitcher   
84   Cleveland Indians           Bard, Josh    302,100         Catcher   
85   Cleveland Indians          Bere, Jason  1,000,000         Pitcher   
86   Cleveland Indians         Blake, Casey    330,000   Third Baseman   
87   Cleveland Indians      Bradley, Milton    314,300      Outfielder   
88   Cleveland Indians  Broussard, Benjamin    303,000   First Baseman   
89   Cleveland Indians         Burks, Ellis  7,166,667      Outfielder   
90   Cleveland Indians         Davis, Jason    301,100         Pitcher   
91   Cleveland Indians        Garcia, Karim    900,000      Outfielder   
92   Cleveland Indians     Gutierrez, Ricky  3,916,667       Shortstop   
93   Cleveland Indians       Hafner, Travis    302,200   First Baseman   
94   Cleveland Indians           Laker, Tim    400,000         Catcher   
95   Cleveland Indians         Lawton, Matt  6,750,000      Outfielder   
96   Cleveland Indians           Lee, Cliff    300,900         Pitcher   
97   Cleveland Indians       McDonald, John    314,400       Shortstop   
98   Cleveland Indians    Mulholland, Terry    500,000         Pitcher   
99   Cleveland Indians        Myette, Aaron    307,500         Pitcher   
100  Cleveland Indians    Phillips, Brandon    300,900       Shortstop   
101  Cleveland Indians         Riske, David    314,000         Pitcher   
102  Cleveland Indians   Rodriguez, Ricardo    302,400         Pitcher   
103  Cleveland Indians         Sabathia, CC  1,100,000         Pitcher   
104  Cleveland Indians         Sadler, Carl    303,200         Pitcher   
105  Cleveland Indians       Santiago, Jose    600,000         Pitcher   
106  Cleveland Indians          Selby, Bill    325,000  Second Baseman   
107  Cleveland Indians       Spencer, Shane    600,000      Outfielder   
108  Cleveland Indians        Traber, Billy    300,000         Pitcher   
109  Cleveland Indians        Vizquel, Omar  5,500,000       Shortstop   
110  Cleveland Indians      Westbrook, Jake    305,500         Pitcher   
111  Cleveland Indians         Wickman, Bob  6,000,000         Pitcher   
112  Cleveland Indians        Wohlers, Mark  2,600,000         Pitcher   

     intTeam  
82   1500000  
83   5125000  
84    302100  
85   1000000  
86    330000  
87    314300  
88    303000  
89   7166667  
90    301100  
91    900000  
92   3916667  
93    302200  
94    400000  
95   6750000  
96    300900  
97    314400  
98    500000  
99    307500  
100   300900  
101   314000  
102   302400  
103  1100000  
104   303200  
105   600000  
106   325000  
107   600000  
108   300000  
109  5500000  
110   305500  
111  6000000  
112  2600000

In [204]:
int_team = pd1["Salary"].apply(lambda x: int("".join(x.split(","))))

In [205]:
pd1["intTeam"] = int_team

In [206]:
clevelandI_team[clevelandI_team["intTeam"] == max(clevelandI_team["intTeam"])] 

Team        Player     Salary    Position  intTeam
89  Cleveland Indians  Burks, Ellis  7,166,667  Outfielder  7166667

### Question3

**What is the average salary of the Outfielder players in Tampa Bay Devil Rays?**

In [207]:
Tampa_outfielders = pd1[(pd1["Position"] == "Outfielder") & (pd1["Team"] == "Tampa Bay Devil Rays")]

In [208]:
Tampa_outfielders

Team           Player     Salary    Position  intTeam
171  Tampa Bay Devil Rays  Baldelli, Rocco    300,000  Outfielder   300000
175  Tampa Bay Devil Rays   Crawford, Carl    300,000  Outfielder   300000
176  Tampa Bay Devil Rays      Grieve, Ben  5,500,000  Outfielder  5500000
182  Tampa Bay Devil Rays       Martin, Al    300,000  Outfielder   300000

In [209]:
Tampa_outfielders["intTeam"].mean()

1600000.0

### Question4
**What is the name of the Catcher earning the lowest salary in the Minnesota Twins Team?**

In [210]:
X = pd1[(pd1["Team"]== "Minnesota Twins") & (pd1["Position"] == "Catcher")]

In [211]:
X

Team          Player   Salary Position  intTeam
232  Minnesota Twins    Lecroy, Matt  312,500  Catcher   312500
238  Minnesota Twins  Pierzynski, AJ  365,000  Catcher   365000
239  Minnesota Twins     Prince, Tom  450,000  Catcher   450000

In [212]:
X=X.sort_values("Salary", ascending = True).iloc[0]

In [213]:
X["Player"]

'Lecroy, Matt'

### Question 5

**What is the total number of players in the Kansas City Royals Team?**

In [214]:
pd5=pd1[(pd1["Team"] == "Kansas City Royals")]

In [215]:
pd5

Team               Player      Salary        Position  \
194  Kansas City Royals      Affeldt, Jeremy     313,000         Pitcher   
195  Kansas City Royals      Asencio, Miguel     314,000         Pitcher   
196  Kansas City Royals      Beltran, Carlos   6,000,000      Outfielder   
197  Kansas City Royals      Berger, Grandon     304,000      Outfielder   
198  Kansas City Royals        Berroa, Angel     302,000       Shortstop   
199  Kansas City Royals        Brown, Dermal     309,500      Outfielder   
200  Kansas City Royals        Bukvich, Ryan     304,500         Pitcher   
201  Kansas City Royals         Carrasco, DJ     300,000         Pitcher   
202  Kansas City Royals       Difelice, Mike     625,000         Catcher   
203  Kansas City Royals       Febles, Carlos     775,000  Second Baseman   
204  Kansas City Royals        George, Chris     303,500         Pitcher   
205  Kansas City Royals      Grimsley, Jason   2,000,000         Pitcher   
206  Kansas City Royals          Harvey, Ken     300,000   First Baseman   
207  Kansas City Royals  Hernandez, Runelvys     305,500         Pitcher   
208  Kansas City Royals         Ibanez, Raul   3,000,000      Outfielder   
209  Kansas City Royals     Johnson, Rontrez     300,000      Outfielder   
210  Kansas City Royals         Lopez, Albie   1,500,000         Pitcher   
211  Kansas City Royals         Lopez, Mendy     300,000       Shortstop   
212  Kansas City Royals      MacDougal, Mike     301,000         Pitcher   
213  Kansas City Royals         May, Darrell     450,000         Pitcher   
214  Kansas City Royals         Mayne, Brent   2,750,000         Catcher   
215  Kansas City Royals           Randa, Joe   4,500,000   Third Baseman   
216  Kansas City Royals       Relaford, Desi     900,000       Shortstop   
217  Kansas City Royals        Sweeney, Mike  11,000,000   First Baseman   
218  Kansas City Royals      Tucker, Michael   2,750,000      Outfielder   
219  Kansas City Royals         Wilson, Kris     311,000         Pitcher   

      intTeam  
194    313000  
195    314000  
196   6000000  
197    304000  
198    302000  
199    309500  
200    304500  
201    300000  
202    625000  
203    775000  
204    303500  
205   2000000  
206    300000  
207    305500  
208   3000000  
209    300000  
210   1500000  
211    300000  
212    301000  
213    450000  
214   2750000  
215   4500000  
216    900000  
217  11000000  
218   2750000  
219    311000

In [216]:
len(pd5["Team"])

26

### Question 6

**How many First Basemen does the Oakland Athletics Team have?**

In [106]:
pd6=pd1[(pd1["Team"] == "Oakland Athletics") & (pd1["Position"] == "First Baseman")]

In [107]:
pd6

Team           Player     Salary       Position  intTeam
329  Oakland Athletics  Durazo, Erubiel  1,065,000  First Baseman  1065000

In [108]:
len(pd6)

1

### Question 7

**How much money does the Texas Rangers Team pay their players each month in total?**

In [115]:
pd7=pd1[(pd1["Team"] == "Texas Rangers")]

In [116]:
pd7

Team              Player      Salary        Position   intTeam
351  Texas Rangers       Blalock, Hank     302,500   Third Baseman    302500
352  Texas Rangers     Clark, Jermaine     300,000  Second Baseman    300000
353  Texas Rangers  Cordero, Francisco     900,000         Pitcher    900000
354  Texas Rangers         Diaz, Einar   1,837,500         Catcher   1837500
355  Texas Rangers       Everett, Carl   9,150,000      Outfielder   9150000
356  Texas Rangers        Fultz, Aaron     600,000         Pitcher    600000
357  Texas Rangers    Garcia, Reynaldo     300,000         Pitcher    300000
358  Texas Rangers     Glanville, Doug   1,000,000      Outfielder   1000000
359  Texas Rangers      Gonzalez, Juan  13,000,000      Outfielder  13000000
360  Texas Rangers        Greene, Todd     750,000         Catcher    750000
361  Texas Rangers        Greer, Rusty   7,000,000      Outfielder   7000000
362  Texas Rangers       Kreuter, Chad     750,000         Catcher    750000
363  Texas Rangers          Lamb, Mike     440,000   Third Baseman    440000
364  Texas Rangers        Lewis, Colby     302,500         Pitcher    302500
365  Texas Rangers        Mench, Kevin     327,500      Outfielder    327500
366  Texas Rangers       Nitkowski, CJ     550,000         Pitcher    550000
367  Texas Rangers    Palmeiro, Rafael   9,000,000   First Baseman   9000000
368  Texas Rangers       Park, Chan Ho  13,000,000         Pitcher  13000000
369  Texas Rangers      Perry, Herbert   1,300,000   Third Baseman   1300000
370  Texas Rangers         Powell, Jay   3,250,000         Pitcher   3250000
371  Texas Rangers     Rodriguez, Alex  22,000,000       Shortstop  22000000
372  Texas Rangers       Sierra, Ruben     600,000      Outfielder    600000
373  Texas Rangers      Teixeira, Mark     750,000   Third Baseman    750000
374  Texas Rangers       Thomson, John   1,300,000         Pitcher   1300000
375  Texas Rangers      Urbina, Ugueth   4,500,000         Pitcher   4500000
376  Texas Rangers      Valdes, Ismael   2,500,000         Pitcher   2500000
377  Texas Rangers    Van Poppel, Todd   2,500,000         Pitcher   2500000
378  Texas Rangers        Yan, Esteban   1,500,000         Pitcher   1500000
379  Texas Rangers       Young, Mike B     415,000       Shortstop    415000
380  Texas Rangers     Zimmerman, Jeff   3,366,667         Pitcher   3366667

In [117]:
pd7["intTeam"].sum()

103491667

### Question 8

**What position does the highest-paid player in the American League (AL) baseball play?**

In [122]:
pd8= pd1[pd1["intTeam"] ==pd1["intTeam"].max()]

In [123]:
pd8

Team           Player      Salary   Position   intTeam
371  Texas Rangers  Rodriguez, Alex  22,000,000  Shortstop  22000000

In [126]:
pd8["Position"]

371    Shortstop
Name: Position, dtype: object

### Question 9
**Which baseball team pays its players the highest salaries in the AL East division?**

In [222]:
pd9 = pd1[pd1["Team"].isin(["Baltimore Orioles", "Boston Red Sox", "New York Yankees", "Tampa Bay Devil Rays", "Toronto Blue Jays"])]

In [223]:
pd9

Team            Player      Salary        Position  \
0        New York Yankees     Acevedo, Juan     900,000         Pitcher   
1        New York Yankees   Anderson, Jason     300,000         Pitcher   
2        New York Yankees    Clemens, Roger  10,100,000         Pitcher   
3        New York Yankees   Contreras, Jose   5,500,000         Pitcher   
4        New York Yankees    Flaherty, John     750,000         Catcher   
..                    ...               ...         ...             ...   
189  Tampa Bay Devil Rays   Shumpert, Terry     300,000  Second Baseman   
190  Tampa Bay Devil Rays       Sosa, Jorge     300,000         Pitcher   
191  Tampa Bay Devil Rays  Valentin, Javier     300,000         Catcher   
192  Tampa Bay Devil Rays     Venafro, Mike     300,000         Pitcher   
193  Tampa Bay Devil Rays  Zambrano, Victor     300,000         Pitcher   

      intTeam  
0      900000  
1      300000  
2    10100000  
3     5500000  
4      750000  
..        ...  
189    300000  
190    300000  
191    300000  
192    300000  
193    300000  

[109 rows x 5 columns]

In [224]:
pdNY = pd9[pd9["Team"] == "New York Yankees"]

In [225]:
pdNY.head()

Team           Player      Salary Position   intTeam
0  New York Yankees    Acevedo, Juan     900,000  Pitcher    900000
1  New York Yankees  Anderson, Jason     300,000  Pitcher    300000
2  New York Yankees   Clemens, Roger  10,100,000  Pitcher  10100000
3  New York Yankees  Contreras, Jose   5,500,000  Pitcher   5500000
4  New York Yankees   Flaherty, John     750,000  Catcher    750000

In [226]:
pdNY = sum(pdNY["intTeam"])

In [227]:
pdNY

152749814

In [228]:
pdBO = pd9[pd9["Team"] == "Baltimore Orioles"]

In [229]:
pdBO.head()

Team         Player      Salary       Position   intTeam
140  Baltimore Orioles  Batista, Tony   6,400,000  Third Baseman   6400000
141  Baltimore Orioles    Bauer, Rick     325,000        Pitcher    325000
142  Baltimore Orioles   Bedard, Erik     300,000        Pitcher    300000
143  Baltimore Orioles  Belle, Albert  13,000,000     Outfielder  13000000
144  Baltimore Orioles   Conine, Jeff   4,250,000     Outfielder   4250000

In [230]:
pdBO = sum(pdBO["intTeam"])

In [231]:
pdBO

73877500

In [232]:
pdBS = pd9[pd9["Team"] == "Boston Red Sox"]

In [233]:
pdBS.head()

Team         Player     Salary    Position  intTeam
55  Boston Red Sox  Burkett, John  5,500,000     Pitcher  5500000
56  Boston Red Sox  Damon, Johnny  7,500,000  Outfielder  7500000
57  Boston Red Sox   Embree, Alan  3,000,000     Pitcher  3000000
58  Boston Red Sox  Fossum, Casey    324,500     Pitcher   324500
59  Boston Red Sox      Fox, Chad    500,000     Pitcher   500000

In [234]:
pdBS = sum(pdBS["intTeam"])

In [235]:
pdBS

99946500

In [236]:
pdTB = pd9[pd9["Team"] == "Tampa Bay Devil Rays"]

In [237]:
pdTB.head()

Team            Player   Salary        Position  intTeam
169  Tampa Bay Devil Rays  Abernathy, Brent  300,000  Second Baseman   300000
170  Tampa Bay Devil Rays  Anderson, Marlon  600,000  Second Baseman   600000
171  Tampa Bay Devil Rays   Baldelli, Rocco  300,000      Outfielder   300000
172  Tampa Bay Devil Rays   Bierbrodt, Nick  300,000         Pitcher   300000
173  Tampa Bay Devil Rays     Carter, Lance  300,000         Pitcher   300000

In [238]:
pdTB = sum(pdTB["intTeam"])

In [239]:
pdTB

19650000

In [240]:
pdTBJ = pd9[pd9["Team"] == "Toronto Blue Jays"]

In [220]:

pdTBJ.head()

Empty DataFrame
Columns: [Team, Player, Salary, Position, intTeam]
Index: []

In [241]:
pdTBJ = sum(pdTBJ["intTeam"])

In [242]:
pdTBJ

0

In [243]:
pd9 = max(pdTBJ,pdTB,pdBS,pdBO,pdNY)

In [244]:
pd9

152749814

##### After calculating the total sum of each teams salary present in the AL East division, we therefore check the maximum amount among all, which results to 152749814, and that proofs the answer to be **New York Yankees team**

### Question 10
**What is the total number of players in the AL Central division?**

In [172]:
pd10 = pd1[pd1["Team"].isin(["Chicago White Sox", "Cleveland Guardians", "Detroit Tigers", "Kansas City Royals", "Minnesota Twins"])]

In [173]:
pd10.head()

Team           Player     Salary    Position  intTeam
194  Kansas City Royals  Affeldt, Jeremy    313,000     Pitcher   313000
195  Kansas City Royals  Asencio, Miguel    314,000     Pitcher   314000
196  Kansas City Royals  Beltran, Carlos  6,000,000  Outfielder  6000000
197  Kansas City Royals  Berger, Grandon    304,000  Outfielder   304000
198  Kansas City Royals    Berroa, Angel    302,000   Shortstop   302000

In [174]:
len(pd10["Player"])

104

After filtering down all the teams in the AL central division, the lenght of the total players is **104**